components/eventHandler.ts

```diff
@@ -719,6 +724,22 @@ function saveEvents(
                 }
                 break
             }
+            case "StartingSuit":
+                session.currentDisguise = event.Value as string
+                break
+            case "ContractFailed":
+                session.timerEnd = event.Timestamp
+                contractFailed(event, session)
+                break
+            case "OpportunityEvents": {
+                const val = (<OpportunityEventsC2SEvent>event).Value
+                const opportunities = userData.Extensions.opportunityprogression
+                if (val.Event === "Completed") {
+                    opportunities[val.RepositoryId] = true
```

Comment:
We'll need to call `writeUserData` after modifying this so it's marked as dirty.